### ISSUE TITLE
Updates and DLCS on v0.4

### ISSUE THREAD

**Issue by werneck9898** [XorTroll/Goldleaf#219]
There is a problem where installing updates and dlc nsps makes atmosphere crash with error code 2168-0002 (0x4a8). Although the update/dlc is applied after rebooting, it makes me having to reboot everytime I install an update or dlc file.

**Comment by XorTroll** [comment]
Yeah, will look into that. Other users seem to face the same issue too.

**Comment by pabermod** [comment]
Same problem happened to me installing an update

**Comment by AlphA-NeoX** [comment]
Same here, although i have something else to add. Goldleaf started crashing when installing updates and DLC's after i downgraded the console from 7.0 to 6.2, i thought it was because i was using Goldleaf v0.3 so i updated to v0.4 and it kept crashing anyways, i had no problems with Goldleaf before the downgrade.

I don't know if there's a correlation between the crashes and downgrade but in my case problems started after downgrading.

**Comment by XorTroll** [comment]
v0.5 will fix this, bug already fixed in local changes.
Will commit later. (more like in a few days, as I have tons of WIP / stubbed stuff)

**Comment by emariz** [comment]
I can confirm this issue in version 0.4, on Atmosphere 0.8.5 and Switch 6.2.0. The application hangs at step 4 of 4 while installing NSP Updates and causes a system crash with error message 2168-0002 (0x4a8). The updates themselves seem to be working correctly.

**Comment by emariz** [comment]
Replying to myself, in case someone else faces this issue. I checked the commits tree and tried four different builds of Goldleaf 0.5:

    f74d9b7 from March 17, the latest available
    69fba6e from March 9, the last of that day's commits
    d212c5a from Feb 25, the last build of that month
    be5ddeb from Feb 16, from commits about a week earlier

While the three most recent builds could install Updates without producing a system crash, build d212c5a was the only version that could also manage installed titles. All the builds from March use a different title layout which, on my system, is unusable. If one clicks on a title within the manager, Goldleaf either shows information about another one (sometimes the one right above, but some others an entirely different one) or just does nothing.

After installing about twenty titles and their updates, I can say that build d212c5a installed them all correctly and did not cause a system crash after an installation. _However_, the application did show a garbled error message when trying to install a title immediately after installing an update. Closing the application and reopening it cleared the message and allowed me to continue installing other titles.

It was a cumbersome to exit the application after installing every update, but at least no system crash was triggered and all titles installed correctly.
